### ISSUE TITLE
Stratify table by the current sorting of a single user-defined categorical column

### ISSUE THREAD

**Issue by mstreit** [Caleydo/mothertable#249]
The sorting of the categorical column should drive the stratification. See GDoc.
Related to #206.

**Comment by bikramkawan** [comment]
Use Case: 
User stratifies by column gender after sorting. The box represent stratification(multiform)
![image](https://cloud.githubusercontent.com/assets/12292333/24079592/b6049c0a-0c8b-11e7-9ad9-81fdd43a1042.png)
Will we now not consider numerical or string as stratification as before right?  Otherwise columns we add after numerical or string has to be split into several chunks which are avoided now in new layout.
 Here is snapshot from older version.
![image](https://cloud.githubusercontent.com/assets/12292333/24079668/50ae7680-0c8d-11e7-800c-c1ad3f8f4847.png)

**Comment by mstreit** [comment]
Yes, we ignore numerical and matrix columns for the stratification.

**Comment by bikramkawan** [comment]
To make  row consistent the sorting of numerical or column will be overridden by the following categorical column:
![image](https://cloud.githubusercontent.com/assets/12292333/24079702/4825e4a2-0c8e-11e7-9332-39441815fea6.png)

**Comment by mstreit** [comment]
I don't understand your last sketch. 
Let me rephrase it: The user can select a single categorical column. We take whatever combined stratification we have there and apply it across all columns.

**Comment by bikramkawan** [comment]
It was just to refresh the previous concept if we take numerical column as a part of stratification which is not valid anymore according to our new concept.

**Comment by mstreit** [comment]
I am not sure about this. Basically the user should get the stratification that he currently sees in this column. This is the only thing a user will understand. If he adds a numerical column and puts it in the hierarchy over the categorical column that he wants to use as a stratifier, then he will end up with many groups (but this corresponds to what he currently sees). If he wants to reduce the number of groups, then he needs to move the column higher in the hierarchy (above the numerical column).

**Comment by bikramkawan** [comment]
![2017-03-19_18-16-56](https://cloud.githubusercontent.com/assets/12292333/24083062/704ae5f8-0cd0-11e7-8b17-932a4ac31f6f.gif)

**Comment by mstreit** [comment]
Could you draw a 2px border around every multiform in the implementation (just for debugging purposes)? This would make it much easier to see what the stratification is.

**Comment by mstreit** [comment]
Or alternatively add the spacing described in #206.

**Comment by bikramkawan** [comment]
Stratified by Ethnicity

![image](https://cloud.githubusercontent.com/assets/12292333/24083152/a34f744e-0cd2-11e7-8b20-839fd099f551.png)

**Comment by mstreit** [comment]
Looks good! I think the spacing can be a bit smaller. How many pixels are this?

**Comment by bikramkawan** [comment]
At the moment its 9px

**Comment by mstreit** [comment]
Oh wait, the matrix seems to be not correctly stratified. There is a split missing in the lower part.

**Comment by mstreit** [comment]
(let's move the spacing discussion to issue #206.

**Comment by mstreit** [comment]
Let's also add the first added categorical column as default stratification (see concept description).

**Comment by katush** [comment]
@bikramkawan you probably know about this, but the current stratification functionality overrides the sorting - this shouldn't happen I think. According to the filter view It should be first sorted by numerical column.
![image](https://cloud.githubusercontent.com/assets/8986080/24104175/c6ec0e94-0d81-11e7-97cb-b61d69deac19.png)

**Comment by bikramkawan** [comment]
Yes. Just fixed.
![image](https://cloud.githubusercontent.com/assets/12292333/24104950/d40aecc4-0d83-11e7-80fb-de7e3403f131.png)
